### PR TITLE
chore: use componentInfo per package

### PR DESCRIPTION
**Pull request by layershifter**
Related #14449.

---

This PR moves `componentInfo` from `@fluentui/docs` package to matching packages. This fixes the issue with unclear build dependencies between packages.

Also removes UMD build, it's broken even from Stardust UI times and is nowhere used.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -3,7 +3,6 @@
 *.info.json
 *.tsbuildinfo
 packages/fluentui/docs/src/componentInfo
```
Comment (layershifter) [inline .gitignore:5]: This should be updated

File: packages/fluentui/react-northstar/gulpfile.ts [hunk 2]
```diff
@@ -4,5 +4,5 @@ import { parallel, series, task } from 'gulp';
 import '../../../gulpfile';
 
 // Redefine test and test:watch to build info json files beforehand
-task('test', series('build:docs:component-info', 'test:jest'));
-task('test:watch', series('build:docs:component-info', parallel('test:jest:watch', 'watch:docs:component-info')));
+task('test', series('build:component-info', 'test:jest'));
```
Comment (dzearing) [inline packages/fluentui/react-northstar/gulpfile.ts:7]: does this still need to happen here if it's happening in `build` ?

Comment (layershifter) [reply packages/fluentui/react-northstar/gulpfile.ts:7]: Yes, as in Northstar there is no requirement to build before test run

File: packages/fluentui/docs/src/components/Sidebar/Sidebar.tsx [hunk 3]
```diff
@@ -22,11 +22,13 @@ import { NavLink, withRouter, RouteComponentProps } from 'react-router-dom';
...
 import { renderSidebarTitle } from './SidebarTitle';
 import config from '../../config';
+import componentInfoContext from '../../utils/componentInfoContext';
 
 type ComponentMenuItem = { displayName: string; type: string };
 
 const pkg = require('@fluentui/react-northstar/package.json');
-const componentMenu: ComponentMenuItem[] = require('../../componentMenu');
+
+const componentMenu: ComponentMenuItem[] = _.sortBy(componentInfoContext.parents, 'displayName');
```
Comment (layershifter) [inline packages/fluentui/docs/src/components/Sidebar/Sidebar.tsx:31]: This change allows to get rid of additional JSON file 🚀

File: scripts/gulp/tasks/docs.ts [hunk 4]
```diff
@@ -212,14 +203,26 @@ task('serve:docs:stop', () => forceClose(server));
...
+          )
+          .pipe(dest('componentInfo', { cwd: pkg.packagePath }));
+      // @ts-ignore
+      internalTask.displayName = 'build:docs:component-info';
+
+      const watcher = watch(pkg.packageJson.gulp?.componentInfo, { cwd: pkg.packagePath }, internalTask);
+
+      watcher.on('add', logWatchAdd);
+      watcher.on('change', logWatchChange);
+    }
```
Comment (layershifter) [inline scripts/gulp/tasks/docs.ts:224]: It will be great to remove this, but running three copies of `watch:component-info` explodes my machine 😭

**Comment by ecraig12345** [comment]
Went ahead and merged this to see if it fixes another issue. Hopefully that's okay (or if it's really bad we can revert).

**Pull request merged**